### PR TITLE
[ONNX] Hardmax support in ONNX importer

### PR DESCRIPTION
**Pull request by tomdol**

### CODE REVIEW

**Comment by postrational** [comment]
Related ONNX tests: https://github.com/NervanaSystems/ngraph-onnx/pull/368

File: src/ngraph/frontend/onnx_import/op/hardmax.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            namespace set_1
+            {
+                NodeVector hardmax(const Node& node)
+                {
+                    const auto input = node.get_ng_inputs().at(0);
+                    const auto& input_shape = input->get_shape();
+                    auto axis = node.get_attribute_value<std::int64_t>("axis", 1);
+
+                    NGRAPH_CHECK(axis >= 0 && axis < input_shape.size(),
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/hardmax.cpp]: Please use `ASSERT_VALID_ARGUMENT` here, to include the node information in the error message.

Please also check if it's possible to use the newer `NODE_VALIDATION_CHECK`  here to replace the older and ONNX-specific `ASSERT_VALID_ARGUMENT`.

File: src/ngraph/frontend/onnx_import/utils/eye.hpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    namespace onnx_import
+    {
+        namespace eye
+        {
+            /// \brief Creates a square identity matrix.
+            ///
+            /// \param[in] n Order of the resulting matrix.
+            ///
+            /// \return A Constant node representing identity matrix with shape (n, n).
+            template <typename T = double>
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/utils/eye.hpp]: Do we need this to be a template?

Comment (tomdol) [reply src/ngraph/frontend/onnx_import/utils/eye.hpp]: The template version can be used for other ops, for example `EyeLike` which should support more types.

File: src/ngraph/frontend/onnx_import/op/hardmax.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "hardmax.hpp"
+#include "exceptions.hpp"
```
[the remark this answers is not in the document]
Comment (tomdol) [reply src/ngraph/frontend/onnx_import/op/hardmax.cpp]: This order is what I got from `make style-apply`. 

When I reorder them alphabetically and run `make style-check` I get an error:
```
-- style error: /home/tdolbnia/projects/clean/ngraph/src/ngraph/frontend/onnx_import/op/hardmax.cpp
CMake Error at /home/tdolbnia/projects/clean/ngraph/cmake/Modules/style_check.cmake:48 (message):
  style errors
```

so I guess I have to leave it as is.

File: src/ngraph/frontend/onnx_import/op/hardmax.cpp (hunk 1 above)
Comment (tomdol) [reply src/ngraph/frontend/onnx_import/op/hardmax.cpp]: `NODE_VALIDATION_CHECK` expects an `ngraph::Node` so I couldn't use it but I've changed the assertion macro to the suggested `ASSERT_VALID_ARGUMENT`

**Pull request merged**